app/src/main/java/test/arctouch/tmdbapp/data/remote/TMDbRemoteData.java

```diff
@@ -0,0 +1,28 @@
+package test.arctouch.tmdbapp.data.remote;
+
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import rx.Observable;
+import test.arctouch.tmdbapp.network.TMDbApi;
+import test.arctouch.tmdbapp.network.model.MovieResponse;
+
+@Singleton
+public class TMDbRemoteData {
```

Comment:
Well, in this case as you said I have already prepared the app for more complexity and I really like to the repository pattern along with SOLID with the principle single responsibility. My propose for this app was every presenter should talk only with the repository. The repository has the responsability to get the data from the `TMDbRemoteData` or in case of error get from `TMDbLocalData`  (which I havent created)  which means data base, shared preferences and etc.